Benchmark doc (#3394)
yugabyte/yugabyte-db: docs/content/latest/benchmark/tpcc.md

---
title: TPCC
linkTitle: TPCC
description: TPCC
image: /images/section_icons/architecture/concepts.png
headcontent: Benchmark YugabyteDB using TPCC.
menu:
  latest:
    identifier: benchmark-tpcc
    parent: benchmark
    weight: 740
aliases:
  - /benchmark/tpcc/
showAsideToc: True
isTocNested: True
---

> 

Reviewer: Lets add a brief description of what this page includes. Something like
"This page will show you how to run TPCC benchmark on YSQL."

Author: Done.

## Step 1. Install Java and Ant.

## Step 2. Clone the TPCC repository.

You can do this by running the following commands.

```sh
cd $HOME
git clone https://github.com/yugabyte/tpcc.git
cd tpcc
```

## Step 3. Compile the code

You can compile the code using the following commands.

```sh

> ant bootstrap

Reviewer: Will prefer to move `cd $HOME/tpcc` to this step.

Author: Removed the compilation part since, we are now using the tar ball based approach.

ant resolve
ant build
```

## Step 4. Start your database
Start the database using steps mentioned here: https://docs.yugabyte.com/latest/quick-start/explore-ysql/.

## Step 5. Configure connection properties

> Set the following connection configurations in yugabyteSQL/db.properties:

Reviewer: What is yugabyteSQL/db.properties? Is this a new file that needs to be created?

Since we already have the config files committed, it'll be easier to just ask users to modify ip address in `config/workload_all` if needed.

Author: Changed the statement.

```sh
<!-- Connection details -->
<dbtype>postgres</dbtype>
<driver>org.postgresql.Driver</driver>
<DBUrl>jdbc:postgresql://<ip>:5433/yugabyte</DBUrl>
<username>yugabyte</username>
<password></password>
<isolation>TRANSACTION_REPEATABLE_READ</isolation>
```

The details of the workloads have already been populated in the sample config present in /config.
The workload descriptor works the same way as it does in the upstream branch and details can be found in the [on-line documentation](https://github.com/oltpbenchmark/oltpbench/wiki).

## Step 6. Running the Benchmark
A utility script (./tpccbenchmark) is provided for running the benchmark. The options are

```
-c,--config &lt;arg&gt;            [required] Workload configuration file
   --clear &lt;arg&gt;             Clear all records in the database for this
                             benchmark
   --create &lt;arg&gt;            Initialize the database for this benchmark
   --dialects-export &lt;arg&gt;   Export benchmark SQL to a dialects file
   --execute &lt;arg&gt;           Execute the benchmark workload

> -h,--help                    Print this help

Reviewer: Formatting looks off here.

Author: This part is from the official OLTPbench repo.
Let me know if you still want me to change it.

Reviewer: Yes please :)

   --histograms              Print txn histograms
   --load &lt;arg&gt;              Load data using the benchmark's data loader
-o,--output &lt;arg&gt;            Output file (default System.out)
   --runscript &lt;arg&gt;         Run an SQL script
-s,--sample &lt;arg&gt;            Sampling window
-v,--verbose                 Display Messages
```

Before you can actually run the workload, you need to "load" the data first.

```sh
./tpccbenchmark -c config/workload_1.xml --create=true --load=true
```

Then, you can run the workload:

```sh
./tpccbenchmark -c config/workload_1.xml --execute=true -s 5 -o outputfile
```

> We can also load and perform the benchmark together as well:

Reviewer: ```suggestion
We can also load and perform the benchmark in a single step:
```

Author: Done.

---
title: TPCC
linkTitle: TPCC
description: TPCC
image: /images/section_icons/architecture/concepts.png
headcontent: Benchmark YugabyteDB using TPCC.
menu:
  latest:
    identifier: benchmark-tpcc
    parent: benchmark
    weight: 740
aliases:
  - /benchmark/tpcc/
showAsideToc: True
isTocNested: True
---

> Following are the steps to be taken to run TPCC against Yugabyte.

Reviewer: ```suggestion
This page shows you how to run TPCC against YugabyteDB.
```
(Since you are using second person pronouns "you" elsewhere, lets keep the start also consistent).

Author: Done.

## Step 1. Download the TPCC binaries.

You can do this by running the following commands.

```sh
cd $HOME
wget https://github.com/yugabyte/tpcc/releases/download/1.0/tpcc.tar.gz
tar -zxvf tpcc.tar.gz
cd tpcc
```

## Step 2. Start your database
Start the database using steps mentioned here: https://docs.yugabyte.com/latest/quick-start/explore-ysql/.

## Step 3. Configure connection properties

> Set the following connection configurations in the workload config file.

Reviewer: It's not clear which file you are referring to:
```suggestion
Set the following connection configurations in the workload config file `config/workload_1.xml`.
```

Author: Done.

```sh
<!-- Connection details -->
<dbtype>postgres</dbtype>
<driver>org.postgresql.Driver</driver>
<DBUrl>jdbc:postgresql://<ip>:5433/yugabyte</DBUrl>
<username>yugabyte</username>
<password></password>
<isolation>TRANSACTION_REPEATABLE_READ</isolation>
```

> The details of the workloads have already been populated in the sample config present in /config.

Reviewer: ```suggestion
The details of the workloads have already been populated in the sample configs present in /config.
```

Author: Done.

The workload descriptor works the same way as it does in the upstream branch and details can be found in the [on-line documentation](https://github.com/oltpbenchmark/oltpbench/wiki).

## Step 4. Running the Benchmark
A utility script (./tpccbenchmark) is provided for running the benchmark. The options are

```
-c,--config &lt;arg&gt;            [required] Workload configuration file
   --clear &lt;arg&gt;             Clear all records in the database for this
                             benchmark
   --create &lt;arg&gt;            Initialize the database for this benchmark
   --dialects-export &lt;arg&gt;   Export benchmark SQL to a dialects file
   --execute &lt;arg&gt;           Execute the benchmark workload
-h,--help                    Print this help
   --histograms              Print txn histograms
   --load &lt;arg&gt;              Load data using the benchmark's data loader
-o,--output &lt;arg&gt;            Output file (default System.out)
   --runscript &lt;arg&gt;         Run an SQL script
-s,--sample &lt;arg&gt;            Sampling window
-v,--verbose                 Display Messages
```

> Before you can actually run the workload, you need to "load" the data first.

Reviewer: ```suggestion
Before starting the the workload, you need to "load" the data first.
```

Author: Done.

```sh
./tpccbenchmark -c config/workload_1.xml --create=true --load=true
```

Then, you can run the workload:

```sh
./tpccbenchmark -c config/workload_1.xml --execute=true -s 5 -o outputfile
```

> We can also load and perform the benchmark in a single step:

Reviewer: ```suggestion
You can also load and run the benchmark in a single step:
```

Author: Done.

```sh
./tpccbenchmark -c config/workload_1.xml --create=true --load=true --execute=true -s 5 -o outputfile
```

The config directory has a few configurations for the various workloads. We can run any of those workloads by changing the configuration file.

```sh
./tpccbenchmark -c config/workload_2.xml --create=true --load=true --execute=true -s 5 -o outputfile

> ```

Reviewer: We should also add a section on interpreting the output similar to:
https://github.com/oltpbenchmark/oltpbench/wiki#output

Author: Done.
